TCC.Lib/Helpers/TaskHelper.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace TCC.Lib.Helpers
+{
+    public static class TaskHelper
+    {
+        private class ParallelizeCore : IDisposable
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Fix this implementation of 'IDisposable' to conform to the dispose pattern. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3881)